pkg/scheduler/framework/types.go

```diff
@@ -93,9 +93,9 @@ type WeightedAffinityTerm struct {
 
 // FitError describes a fit error of a pod.
 type FitError struct {
-	Pod                   *v1.Pod
-	NumAllNodes           int
-	FilteredNodesStatuses NodeToStatusMap
+	Pod         *v1.Pod
+	NumAllNodes int
+	Diagnosis   *Diagnosis
```

Comment:
this just contains 2 maps, so no need to make this a pointer. Saves one instantiation.

Reply:
Changed to a value type.